plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java

```diff
@@ -18,112 +18,45 @@
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
-import org.gradle.api.execution.TaskExecutionGraph;
 import org.gradle.api.plugins.BasePlugin;
-import org.gradle.api.plugins.JavaBasePlugin;
+import org.gradle.util.GradleVersion;
 
 import com.diffplug.spotless.SpotlessCache;
 
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import groovy.lang.Closure;
-
 public class SpotlessPlugin implements Plugin<Project> {
 	SpotlessExtension spotlessExtension;
 
-	static final String EXTENSION = "spotless";
-	static final String CHECK = "Check";
-	static final String APPLY = "Apply";
-
-	private static final String TASK_GROUP = "Verification";
-	private static final String CHECK_DESCRIPTION = "Checks that sourcecode satisfies formatting steps.";
-	private static final String APPLY_DESCRIPTION = "Applies code formatting steps to sourcecode in-place.";
-	private static final String FILES_PROPERTY = "spotlessFiles";
-
 	@Override
 	public void apply(Project project) {
 		// make sure there's a `clean` task
 		project.getPlugins().apply(BasePlugin.class);
 
 		// setup the extension
-		spotlessExtension = project.getExtensions().create(EXTENSION, SpotlessExtension.class, project);
+		spotlessExtension = project.getExtensions().create(SpotlessExtension.EXTENSION, SpotlessExtension.class, project);
+
+		// clear spotless' cache when the user does a clean
+		Task clean = project.getTasks().getByName(BasePlugin.CLEAN_TASK_NAME);
+		clean.doLast(unused -> SpotlessCache.clear());
 
-		// after the project has been evaluated, configure the check and format tasks per source set
-		project.afterEvaluate(this::createTasks);
+		project.afterEvaluate(unused -> {
```

Comment:
Is there value in doing this in an `afterEvaluate`?

Reply:
I think so.  People can have multiple `spotless` blocks (such as a default one in a `subprojects` block in the the root project, then making little changes in a specific subproject), so we have to wait until the script has finished running to know if `enforceCheck` is true or not.  It's the same reason we do `afterEvaluate` here: https://github.com/diffplug/spotless/blob/e97a580229601ec031c551c46a23217245117888/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessExtension.java#L244

We used to do all the config in `afterEvaluate`, the main change in the first commit is to create the unconfigured tasks immediately, because it makes it easier for users to connect them with other tasks to not have to do those connections in an `afterEvaluate` block.